<?xml version="1.0"?>
<robot name="CHIDORI"
       xmlns:xi="http://www.w3.org/2001/XInclude"
       >
  <link name="BODY">
    <visual>
      <origin xyz="0 0 1" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/BODY_mesh.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 1" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/BODY_mesh.dae" scale="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <mass value="14.3177" />
      <origin xyz="-0.0394115 0.00222456 0.122654" rpy="0 -0 0"/>
      <inertia ixx="0.458867" ixy="0.005282" ixz="-0.029983" iyy="0.490286" iyz="0.00532" izz="0.311201"/>
    </inertial>
  </link>
  <gazebo reference="BODY">
    <mu1>0.9</mu1>
    <mu2>0.9</mu2>
  </gazebo>
  <link name="LLEG_LINK0">
    <visual>
      <origin xyz="0 0.1 0.0095" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/LLEG_LINK0_mesh.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0.1 0.0095" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/LLEG_LINK0_mesh.dae" scale="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <mass value="2.70445" />
      <origin xyz="-0.0232272 -0.00636194 -0.0791829" rpy="0 -0 0"/>
      <inertia ixx="0.025545" ixy="0.000389" ixz="0.006217" iyy="0.029606" iyz="0.000979" izz="0.007544"/>
    </inertial>
  </link>
  <gazebo reference="LLEG_LINK0">
    <mu1>0.9</mu1>
    <mu2>0.9</mu2>
  </gazebo>
  <link name="LLEG_LINK1">
    <visual>
      <origin xyz="-0.002 0 -0.135" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/LLEG_LINK1_mesh.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="-0.002 0 -0.135" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/LLEG_LINK1_mesh.dae" scale="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <mass value="0.919687" />
      <origin xyz="-0.00472858 0.00492455 -4e-09" rpy="0 -0 0"/>
      <inertia ixx="0.000969" ixy="3e-06" ixz="0" iyy="0.000749" iyz="0" izz="0.001353"/>
    </inertial>
  </link>
  <gazebo reference="LLEG_LINK1">
    <mu1>0.9</mu1>
    <mu2>0.9</mu2>
  </gazebo>
  <link name="LLEG_LINK2">
    <visual>
      <origin xyz="0 0 0" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/LLEG_LINK2_mesh.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/LLEG_LINK2_mesh.dae" scale="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <mass value="5.50657" />
      <origin xyz="0.00310485 0.0260996 -0.18445" rpy="0 -0 0"/>
      <inertia ixx="0.268042" ixy="-3.9e-05" ixz="-3.9e-05" iyy="0.26448" iyz="-3.9e-05" izz="0.022178"/>
    </inertial>
  </link>
  <gazebo reference="LLEG_LINK2">
    <mu1>0.9</mu1>
    <mu2>0.9</mu2>
  </gazebo>
  <link name="LLEG_LINK3">
    <visual>
      <origin xyz="0 0 -0.35" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/LLEG_LINK3_mesh.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 -0.35" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/LLEG_LINK3_mesh.dae" scale="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <mass value="2.87661" />
      <origin xyz="0.0273698 0.0166011 -0.173694" rpy="0 -0 0"/>
      <inertia ixx="0.12516" ixy="0.000859" ixz="-0.009489" iyy="0.127382" iyz="-0.008796" izz="0.009192"/>
    </inertial>
  </link>
  <gazebo reference="LLEG_LINK3">
    <mu1>0.9</mu1>
    <mu2>0.9</mu2>
  </gazebo>
  <link name="LLEG_LINK4">
    <visual>
      <origin xyz="0 0 -0.35" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/LLEG_LINK4_mesh.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 -0.35" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/LLEG_LINK4_mesh.dae" scale="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <mass value="0.840764" />
      <origin xyz="-0.00992223 0.00249274 -0.00685913" rpy="0 -0 0"/>
      <inertia ixx="0.000887" ixy="-1.8e-05" ixz="2.2e-05" iyy="0.001261" iyz="2e-05" izz="0.000978"/>
    </inertial>
  </link>
  <gazebo reference="LLEG_LINK4">
    <mu1>0.9</mu1>
    <mu2>0.9</mu2>
  </gazebo>
  <link name="LLEG_LINK5">
    <visual>
      <origin xyz="0 0 -0.04" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/LLEG_LINK5_mesh.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 -0.04" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/LLEG_LINK5_mesh.dae" scale="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <mass value="1.8376" />
      <origin xyz="0.00160778 0.0021038 -0.0656516" rpy="0 -0 0"/>
      <inertia ixx="0.012599" ixy="5.7e-05" ixz="-0.000602" iyy="0.013954" iyz="-0.000404" izz="0.005494"/>
    </inertial>
  </link>
  <gazebo reference="LLEG_LINK5">
    <mu1>0.9</mu1>
    <mu2>0.9</mu2>
  </gazebo>
  <link name="RLEG_LINK0">
    <visual>
      <origin xyz="0 -0.1 0.0095" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/RLEG_LINK0_mesh.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 -0.1 0.0095" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/RLEG_LINK0_mesh.dae" scale="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <mass value="2.70137" />
      <origin xyz="-0.0232979 0.00636922 -0.0791194" rpy="0 -0 0"/>
      <inertia ixx="0.025516" ixy="-0.000389" ixz="0.006233" iyy="0.029557" iyz="-0.000979" izz="0.007553"/>
    </inertial>
  </link>
  <gazebo reference="RLEG_LINK0">
    <mu1>0.9</mu1>
    <mu2>0.9</mu2>
  </gazebo>
  <link name="RLEG_LINK1">
    <visual>
      <origin xyz="-0.002 0 -0.135" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/RLEG_LINK1_mesh.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="-0.002 0 -0.135" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/RLEG_LINK1_mesh.dae" scale="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <mass value="0.939505" />
      <origin xyz="-0.0051582 -0.00542379 4e-09" rpy="0 -0 0"/>
      <inertia ixx="0.001014" ixy="-3e-06" ixz="0" iyy="0.000787" iyz="0" izz="0.001419"/>
    </inertial>
  </link>
  <gazebo reference="RLEG_LINK1">
    <mu1>0.9</mu1>
    <mu2>0.9</mu2>
  </gazebo>
  <link name="RLEG_LINK2">
    <visual>
      <origin xyz="0 0 0" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/RLEG_LINK2_mesh.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/RLEG_LINK2_mesh.dae" scale="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <mass value="5.50722" />
      <origin xyz="0.00310486 -0.0260835 -0.184418" rpy="0 -0 0"/>
      <inertia ixx="0.268054" ixy="3.9e-05" ixz="-0.012861" iyy="0.26452" iyz="0.024989" izz="0.02221"/>
    </inertial>
  </link>
  <gazebo reference="RLEG_LINK2">
    <mu1>0.9</mu1>
    <mu2>0.9</mu2>
  </gazebo>
  <link name="RLEG_LINK3">
    <visual>
      <origin xyz="0 0 -0.35" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/RLEG_LINK3_mesh.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 -0.35" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/RLEG_LINK3_mesh.dae" scale="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <mass value="2.88639" />
      <origin xyz="0.027277 -0.0167204 -0.173091" rpy="0 -0 0"/>
      <inertia ixx="0.125176" ixy="-0.000859" ixz="-0.009489" iyy="0.127376" iyz="0.008798" izz="0.009222"/>
    </inertial>
  </link>
  <gazebo reference="RLEG_LINK3">
    <mu1>0.9</mu1>
    <mu2>0.9</mu2>
  </gazebo>
  <link name="RLEG_LINK4">
    <visual>
      <origin xyz="0 0 -0.35" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/RLEG_LINK4_mesh.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 -0.35" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/RLEG_LINK4_mesh.dae" scale="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <mass value="0.840762" />
      <origin xyz="-0.00992232 -0.00249244 -0.00685907" rpy="0 -0 0"/>
      <inertia ixx="0.000887" ixy="1.8e-05" ixz="2.2e-05" iyy="0.001261" iyz="-2e-05" izz="0.000978"/>
    </inertial>
  </link>
  <gazebo reference="RLEG_LINK4">
    <mu1>0.9</mu1>
    <mu2>0.9</mu2>
  </gazebo>
  <link name="RLEG_LINK5">
    <visual>
      <origin xyz="0 0 -0.04" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/RLEG_LINK5_mesh.dae" scale="1 1 1" />
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 -0.04" rpy="0 -0 0"/>
      <geometry>
        <mesh filename="package://hrpsys_ros_bridge_tutorials/models/CHIDORI_meshes/RLEG_LINK5_mesh.dae" scale="1 1 1" />
      </geometry>
    </collision>
    <inertial>
      <mass value="1.84943" />
      <origin xyz="0.00158469 -0.00242358 -0.0657049" rpy="0 -0 0"/>
      <inertia ixx="0.012665" ixy="-9e-05" ixz="-0.0006" iyy="0.014019" iyz="0.000444" izz="0.005494"/>
    </inertial>
  </link>
  <gazebo reference="RLEG_LINK5">
    <mu1>0.9</mu1>
    <mu2>0.9</mu2>
  </gazebo>
  <joint name="LLEG_JOINT0" type="revolute">
    <parent link="BODY"/>
    <child  link="LLEG_LINK0"/>
    <origin xyz="0 0.1 0.0095" rpy="0 -0 0 "/>
    <axis   xyz="0 0 1"/>
    <limit lower="-1.74533" upper="1.74533" effort="219.998" velocity="17.4444" />
    <dynamics damping="0.2" friction="0" />
  </joint>
  <transmission type="pr2_mechanism_model/SimpleTransmission" name="LLEG_JOINT0_trans" >
    <actuator name="LLEG_JOINT0_motor" />
    <joint name="LLEG_JOINT0" />
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  <gazebo reference="LLEG_JOINT0">
    <cfmDamping>0.4</cfmDamping>
  </gazebo>
  <joint name="LLEG_JOINT1" type="revolute">
    <parent link="LLEG_LINK0"/>
    <child  link="LLEG_LINK1"/>
    <origin xyz="-0.002 0 -0.135" rpy="0 -0 0 "/>
    <axis   xyz="1 0 0"/>
    <limit lower="-0.523599" upper="0.785398" effort="449.998" velocity="17.4444" />
    <dynamics damping="0.2" friction="0" />
  </joint>
  <transmission type="pr2_mechanism_model/SimpleTransmission" name="LLEG_JOINT1_trans" >
    <actuator name="LLEG_JOINT1_motor" />
    <joint name="LLEG_JOINT1" />
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  <gazebo reference="LLEG_JOINT1">
    <cfmDamping>0.4</cfmDamping>
  </gazebo>
  <joint name="LLEG_JOINT2" type="revolute">
    <parent link="LLEG_LINK1"/>
    <child  link="LLEG_LINK2"/>
    <origin xyz="0 0 0" rpy="0 -0 0 "/>
    <axis   xyz="0 1 0"/>
    <limit lower="-2.18166" upper="0.785398" effort="449.998" velocity="17.4444" />
    <dynamics damping="0.2" friction="0" />
  </joint>
  <transmission type="pr2_mechanism_model/SimpleTransmission" name="LLEG_JOINT2_trans" >
    <actuator name="LLEG_JOINT2_motor" />
    <joint name="LLEG_JOINT2" />
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  <gazebo reference="LLEG_JOINT2">
    <cfmDamping>0.4</cfmDamping>
  </gazebo>
  <joint name="LLEG_JOINT3" type="revolute">
    <parent link="LLEG_LINK2"/>
    <child  link="LLEG_LINK3"/>
    <origin xyz="0 0 -0.35" rpy="0 -0 0 "/>
    <axis   xyz="0 1 0"/>
    <limit lower="0" upper="2.53073" effort="449.998" velocity="17.4444" />
    <dynamics damping="0.2" friction="0" />
  </joint>
  <transmission type="pr2_mechanism_model/SimpleTransmission" name="LLEG_JOINT3_trans" >
    <actuator name="LLEG_JOINT3_motor" />
    <joint name="LLEG_JOINT3" />
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  <gazebo reference="LLEG_JOINT3">
    <cfmDamping>0.4</cfmDamping>
  </gazebo>
  <joint name="LLEG_JOINT4" type="revolute">
    <parent link="LLEG_LINK3"/>
    <child  link="LLEG_LINK4"/>
    <origin xyz="0 0 -0.35" rpy="0 -0 0 "/>
    <axis   xyz="0 1 0"/>
    <limit lower="-1.5708" upper="1.5708" effort="219.998" velocity="17.4444" />
    <dynamics damping="0.2" friction="0" />
  </joint>
  <transmission type="pr2_mechanism_model/SimpleTransmission" name="LLEG_JOINT4_trans" >
    <actuator name="LLEG_JOINT4_motor" />
    <joint name="LLEG_JOINT4" />
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  <gazebo reference="LLEG_JOINT4">
    <cfmDamping>0.4</cfmDamping>
  </gazebo>
  <joint name="LLEG_JOINT5" type="revolute">
    <parent link="LLEG_LINK4"/>
    <child  link="LLEG_LINK5"/>
    <origin xyz="0 0 -0.04" rpy="0 -0 0 "/>
    <axis   xyz="1 0 0"/>
    <limit lower="-1.0472" upper="1.0472" effort="220.457" velocity="17.4444" />
    <dynamics damping="0.2" friction="0" />
  </joint>
  <transmission type="pr2_mechanism_model/SimpleTransmission" name="LLEG_JOINT5_trans" >
    <actuator name="LLEG_JOINT5_motor" />
    <joint name="LLEG_JOINT5" />
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  <gazebo reference="LLEG_JOINT5">
    <cfmDamping>0.4</cfmDamping>
  </gazebo>
  <joint name="RLEG_JOINT0" type="revolute">
    <parent link="BODY"/>
    <child  link="RLEG_LINK0"/>
    <origin xyz="0 -0.1 0.0095" rpy="0 -0 0 "/>
    <axis   xyz="0 0 1"/>
    <limit lower="-1.74533" upper="1.74533" effort="219.998" velocity="17.4444" />
    <dynamics damping="0.2" friction="0" />
  </joint>
  <transmission type="pr2_mechanism_model/SimpleTransmission" name="RLEG_JOINT0_trans" >
    <actuator name="RLEG_JOINT0_motor" />
    <joint name="RLEG_JOINT0" />
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  <gazebo reference="RLEG_JOINT0">
    <cfmDamping>0.4</cfmDamping>
  </gazebo>
  <joint name="RLEG_JOINT1" type="revolute">
    <parent link="RLEG_LINK0"/>
    <child  link="RLEG_LINK1"/>
    <origin xyz="-0.002 0 -0.135" rpy="0 -0 0 "/>
    <axis   xyz="1 0 0"/>
    <limit lower="-0.785398" upper="0.523599" effort="449.998" velocity="17.4444" />
    <dynamics damping="0.2" friction="0" />
  </joint>
  <transmission type="pr2_mechanism_model/SimpleTransmission" name="RLEG_JOINT1_trans" >
    <actuator name="RLEG_JOINT1_motor" />
    <joint name="RLEG_JOINT1" />
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  <gazebo reference="RLEG_JOINT1">
    <cfmDamping>0.4</cfmDamping>
  </gazebo>
  <joint name="RLEG_JOINT2" type="revolute">
    <parent link="RLEG_LINK1"/>
    <child  link="RLEG_LINK2"/>
    <origin xyz="0 0 0" rpy="0 -0 0 "/>
    <axis   xyz="0 1 0"/>
    <limit lower="-2.0944" upper="0.785398" effort="449.998" velocity="17.4444" />
    <dynamics damping="0.2" friction="0" />
  </joint>
  <transmission type="pr2_mechanism_model/SimpleTransmission" name="RLEG_JOINT2_trans" >
    <actuator name="RLEG_JOINT2_motor" />
    <joint name="RLEG_JOINT2" />
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  <gazebo reference="RLEG_JOINT2">
    <cfmDamping>0.4</cfmDamping>
  </gazebo>
  <joint name="RLEG_JOINT3" type="revolute">
    <parent link="RLEG_LINK2"/>
    <child  link="RLEG_LINK3"/>
    <origin xyz="0 0 -0.35" rpy="0 -0 0 "/>
    <axis   xyz="0 1 0"/>
    <limit lower="0" upper="2.53073" effort="449.998" velocity="17.4444" />
    <dynamics damping="0.2" friction="0" />
  </joint>
  <transmission type="pr2_mechanism_model/SimpleTransmission" name="RLEG_JOINT3_trans" >
    <actuator name="RLEG_JOINT3_motor" />
    <joint name="RLEG_JOINT3" />
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  <gazebo reference="RLEG_JOINT3">
    <cfmDamping>0.4</cfmDamping>
  </gazebo>
  <joint name="RLEG_JOINT4" type="revolute">
    <parent link="RLEG_LINK3"/>
    <child  link="RLEG_LINK4"/>
    <origin xyz="0 0 -0.35" rpy="0 -0 0 "/>
    <axis   xyz="0 1 0"/>
    <limit lower="-1.5708" upper="1.5708" effort="219.998" velocity="17.4444" />
    <dynamics damping="0.2" friction="0" />
  </joint>
  <transmission type="pr2_mechanism_model/SimpleTransmission" name="RLEG_JOINT4_trans" >
    <actuator name="RLEG_JOINT4_motor" />
    <joint name="RLEG_JOINT4" />
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  <gazebo reference="RLEG_JOINT4">
    <cfmDamping>0.4</cfmDamping>
  </gazebo>
  <joint name="RLEG_JOINT5" type="revolute">
    <parent link="RLEG_LINK4"/>
    <child  link="RLEG_LINK5"/>
    <origin xyz="0 0 -0.04" rpy="0 -0 0 "/>
    <axis   xyz="1 0 0"/>
    <limit lower="-1.0472" upper="1.0472" effort="220.457" velocity="17.4444" />
    <dynamics damping="0.2" friction="0" />
  </joint>
  <transmission type="pr2_mechanism_model/SimpleTransmission" name="RLEG_JOINT5_trans" >
    <actuator name="RLEG_JOINT5_motor" />
    <joint name="RLEG_JOINT5" />
    <mechanicalReduction>1</mechanicalReduction>
  </transmission>
  <gazebo reference="RLEG_JOINT5">
    <cfmDamping>0.4</cfmDamping>
  </gazebo>
</robot>
